quill-core/src/main/scala/io/getquill/context/mirror/MirrorContextMacro.scala

```diff
@@ -17,25 +15,30 @@ class MirrorContextMacro(val c: MacroContext) extends ContextMacro {
   override protected def prepare(ast: Ast, params: List[Ident]) =
     IsDynamic(ast) match {
       case false =>
+        val returning = CollectAst(ast) {
```

Comment:
Not really, the query `q.insert.returning(_.1)`, which is transformed to something like `((p1,p2..) => q.insert(_.a -> p1, _.b -> p2...)).apply(p1,p2,p3)` produces the following AST `AssignedAction(FunctionApply(Returning(action, returning), values), assignments)`